docs/localization.md

```diff
@@ -0,0 +1,47 @@
+# Localization
+
+## When should you localize strings?
+
+Any time you add human-readable text to the add-on, whether it is plain text, labels, or text only available to screen-readers, you should localize those strings.
+
+## How to localize strings
+
+First, add the new string to `locales/en-US/strings.properties`. You do NOT need to add strings to other language files.
```

Comment:
Maybe at minimum a note about picking a string identifier as preferring more verbose with a "LOCALIZATION NOTE" as localizers might not be able to run the add-on to see how the strings are used, so whatever context you can provide will be helpful. Link to a lot more details: https://developer.mozilla.org/en-US/docs/Mozilla/Localization/Localization_content_best_practices